tools/common/Application.cs

```diff
@@ -1104,5 +1184,49 @@ public void SetObjectiveCExceptionMode ()
 				break;
 			}
 		}
+
+		public bool IsInterpreted (string assembly)
+		{
+			// IsAOTCompiled and IsInterpreted are not opposites: mscorlib.dll can be both.
+			if (!UseInterpreter)
+				return false;
+
+			// Go through the list of assemblies to interpret in reverse order,
+			// so that the last option passed to mtouch takes precedence.
+			for (int i = InterpretedAssemblies.Count - 1; i >= 0; i--) {
+				var opt = InterpretedAssemblies [i];
+				if (opt == "all")
+					return true;
+				else if (opt == "-all")
+					return false;
+				else if (opt == assembly)
+					return true;
+				else if (opt [0] == '-' && opt.Substring (1) == assembly)
+					return false;
+			}
+
+			// There's an implicit 'all' at the start of the list.
+			return true;
+		}
+
+		public bool IsAOTCompiled (string assembly)
```

Comment:
there's likely a context here for the call to return the correct value
IOW we support JIT (Mac / sim) which are neither interpreted nor AOTed

Reply:
it comes from `Application.mtouch.cs` and while it was already not great (sim) it's even more confusing now that it's shared with Mac.

Reply:
I'm not quite sure which changes you want here: I can add comments to the code, and/or add/handle different cases (i.e. add a conditional and return `false` for Xamarin.Mac for instance). Or did you have anything else in mind?

Reply:
Not 💯% sure myself :-) I'll review the PR again once I complete today's monitoring. The main problem is that we're merging code from two tools into a single one. That's fine except when the code is specific to one product since we lose this _limitation_.

* minimally updated comments so that code does not get reused for XM at some point
* maybe a `#if MONOMAC throw NotImplementedException (); #endif` so the comment can't be missed
* ideally something more OO would be used - but that would be larger changes :(